commands/checkout.go

```diff
@@ -74,6 +82,11 @@ func transformCheckoutArgs(args *Args) error {
 		return err
 	}
 
+	if idx := args.IndexOfParam("--no-branch"); idx >= 0 {
```

Comment:
hub preserves all given flags by default, even non-standard ones. Because git doesn't recognize this flag, we have to remove it. You're doing all right here.